tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -656,6 +656,226 @@ def test_log_null_metric(self):
             warnings.resetwarnings()
         assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
 
+    def test_log_metrics(self):
+
+        run = self._run_factory()
+        run_id = run.info.run_id
+        metric_name = "test-metric-1"
+        metric_tuples = [
+            # Parameters are (key, value, timestamp, step)
+            (metric_name, 10, 0, 0),
+            (metric_name, 20, 1, 1),
+            (metric_name, 40, 3, 3),  # last step, should be stored as last metric
+            (metric_name, 50, 2, 2),
+        ]
+
+        metric_entities = [Metric(*metric_tuple) for metric_tuple in metric_tuples]
+        self.store.log_metrics(run_id, metric_entities)
+
+        metric_histories = self.store.get_metric_history(run_id, metric_name)
+        assert len(metric_histories) == 4
+
+        metrics = [(m.key, m.value, m.timestamp, m.step) for m in metric_histories]
+        assert set(metrics) == set(metric_tuples)
+
+        run_data = self.store.get_run(run_id).data
+        run_metrics = run_data.metrics
+
+        assert len(run_metrics) == 1
+        assert run_metrics[metric_name] == 40
+
+    def test_log_metrics_deactivated_batch_mode(self):
```

Comment:
i don't see the deactivation logic. looks like this test is exactly the same as above

Reply:
Batch mode is deactivate because there are two metrics with same value, ts, step. In this way, the integrity check is tested.
In this test, the comment was missing which I added now to the definition of the metrics.